integration/hsm/helpers.go

```diff
@@ -111,17 +109,43 @@ func (t *teleportService) waitForNewProcess(ctx context.Context) error {
 	return nil
 }
 
+func (t *teleportService) waitForEvent(ctx context.Context, event string) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	waitForEventErr := make(chan error)
+	go func() {
+		_, err := t.process.WaitForEvent(ctx, event)
+		select {
+		case waitForEventErr <- err:
+		case <-ctx.Done():
+		}
+	}()
+	select {
+	case err := <-waitForEventErr:
+		return trace.Wrap(err)
+	case err := <-t.errorChannel:
+		if err != nil {
+			return trace.Wrap(err, "process unexpectedly exited while waiting for event %s", event)
+		}
+		return trace.Errorf("process unexpectedly exited while waiting for event %s", event)
```

Comment:
Isn't `err` nil in this case? Why are we returning an error?

Reply:
Yes it's nil, but the only thing written to this `errorChannel` is the result of `service.Run` which should not return until the service is closed. It should not even return when the process is just reloading, only when it actually exits. Even if it somehow returned a nil error, in this function we are waiting for an event, if the service just exited before we got the event then that's an error